src/window.rs

```diff
@@ -710,11 +726,30 @@ pub trait Window: AsAny + Send + Sync {
     ///
     /// ## Platform-specific
     ///
-    /// - **iOS:** Returns the [`PhysicalSize`] of the window in screen space coordinates.
     /// - **Web:** Returns the size of the canvas element. _Note: this returns the same value as
     ///   [`Window::surface_size`]._
     fn outer_size(&self) -> PhysicalSize<u32>;
 
+    /// The area of the surface that is unobstructed.
+    ///
+    /// On some devices, especially mobile devices, the screen is not a perfect rectangle, and may
+    /// have rounded corners, notches, bezels, and so on. When drawing your content, you usually
+    /// want to draw your background and other such unimportant content on the entire surface, while
+    /// you will want to restrict important content such as text, interactable or visual indicators
+    /// to the part of the screen that is actually visible; for this, you use the safe area.
+    ///
+    /// The safe area is a rectangle that is defined relative to the origin at the top-left corner
+    /// of the surface, and the size extending downwards to the right. The area will not extend
+    /// beyond [the bounds of the surface][Window::surface_size].
+    ///
+    /// If the entire content of the surface is visible, this returns `((0, 0), surface_size)`.
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **Android / Orbital / Wayland / Web / Windows / X11:** Unimplemented, returns `((0, 0),
+    ///   surface_size)`.
+    fn safe_area(&self) -> (PhysicalPosition<u32>, PhysicalSize<u32>);
```

Comment:
The position and size is generally better, since it's easier to forward stuff like that into the e.g. `glViewport` stuff, since you generally want an _offset_ and not left`/right`.

Reply:
I think I prefer returning a type instead of a tuple.
WDYT?

Reply:
Have changed `safe_area` to return the insets instead